.github/workflows/release.yaml

```diff
@@ -1,39 +1,114 @@
-name: release
+# This code is borrowed from https://github.com/kubernetes-sigs/cluster-api/blob/main/.github/workflows/release.yaml
+name: Create Release
 
 on:
   push:
-    # Sequence of patterns matched against refs/tags
-    tags:
-    - "v*"
+    branches:
+      - main
+    paths:
+      - 'CHANGELOG/*.md'
 
-permissions: {}
+permissions:
+  contents: write # Allow to push a tag, create a release branch and publish a draft release.
```

Comment:
Do add it to top-level permissions. Put it into specific job that needs that permission.

Reply:
there are 2 jobs
- push_release_tags
- release
And both need this permission. 
Can steps have different permissions?

Reply:
Yes. Put minimum per job.

Adding to top-level permissions leads to 
1. excessive permissions when job needs differ
2. applies those top-level permissions to new jobs added which might not need any

Permissions is thus best applied per job.

Reply:
done